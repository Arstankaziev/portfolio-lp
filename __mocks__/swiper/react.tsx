import React from 'react'

const SwiperSlide: React.FC = ({ children }) => <>{children}</>
const Swiper: React.FC = ({ children }) => <>{children}</>

export { SwiperSlide, Swiper }
