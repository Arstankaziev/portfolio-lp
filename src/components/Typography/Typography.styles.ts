import styled from 'styled-components'

export const Heading1 = styled.h1`
   font-size: 2.375rem;
   font-weight: 600;
   margin-bottom: 20px;
`
export const Heading2 = styled.h2`
   font-size: 2rem;
   line-height: 2.875rem;
   font-weight: 700;
`
export const Heading3 = styled.h3`
   font-size: 1.5rem;
   font-weight: 500;
`
export const Body = styled.p`
   font-size: 1.5rem;
   line-height: 170%;
   font-weight: 400;
`
export const TextButton = styled.span`
   font-size: 1.125rem;
   font-weight: 400;
`
export const LabelButton = styled.span`
   font-size: 1rem;
   font-weight: 400;
   line-height: 130%;
`

export const Chip = styled.span`
   font-size: 0.875rem;
   line-height: 170%;
   font-weight: 400;
`
export const Label = styled.span`
   font-size: 0.75rem;
   font-weight: 400;
`
