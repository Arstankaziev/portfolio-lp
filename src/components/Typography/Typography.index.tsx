import { ReactNode } from 'react'
import * as S from './Typography.styles'

type Props = {
   type:
      | 'heading-1'
      | 'heading-2'
      | 'heading-3'
      | 'body'
      | 'text-button'
      | 'button'
      | 'chip'
      | 'label'
   children: ReactNode
}

const Typography = ({ type, children }: Props) => {
   switch (type) {
      case 'heading-1':
         return <S.Heading1>{children}</S.Heading1>
      case 'heading-2':
         return <S.Heading2>{children}</S.Heading2>
      case 'heading-3':
         return <S.Heading3>{children}</S.Heading3>
      case 'body':
         return <S.Body>{children}</S.Body>
      case 'text-button':
         return <S.TextButton>{children}</S.TextButton>
      case 'button':
         return <S.LabelButton>{children}</S.LabelButton>
      case 'chip':
         return <S.Chip>{children}</S.Chip>
      case 'label':
         return <S.Label>{children}</S.Label>
      default:
         return <></>
   }
}

export default Typography
