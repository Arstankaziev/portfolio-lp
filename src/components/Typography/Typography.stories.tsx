import Typography from './Typography.index'
import { Meta, Story } from '@storybook/react'

export default {
   title: 'Typography',
   component: Typography
} as Meta

export const Basic: Story = (args) => <Typography {...args} />
