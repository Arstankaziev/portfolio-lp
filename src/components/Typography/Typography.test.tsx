import Typography from './Typography.index'
import { render, screen } from '@testing-library/react'

describe('<Typography />', () => {
   it('should render all headings with correct tags', () => {
      render(
         <>
            <Typography type={'heading-1'} color="light">
               heading1
            </Typography>
            <Typography type={'heading-2'} color="light">
               heading2
            </Typography>
            <Typography type={'heading-3'} color="light">
               heading3
            </Typography>
         </>
      )

      expect(
         screen.getByRole('heading', { name: /heading1/i })
      ).toMatchSnapshot()
      expect(
         screen.getByRole('heading', { name: /heading2/i })
      ).toMatchSnapshot()
      expect(
         screen.getByRole('heading', { name: /heading3/i })
      ).toMatchSnapshot()
   })
})
