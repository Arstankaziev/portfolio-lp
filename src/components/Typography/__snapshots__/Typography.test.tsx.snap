// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography /> should render all headings with correct tags 1`] = `
.c0 {
  font-size: 2.375rem;
  color: #FBFBFB;
  font-weight: 600;
  margin-bottom: 20px;
}

<h1
  class="c0"
>
  heading1
</h1>
`

exports[`<Typography /> should render all headings with correct tags 2`] = `
.c0 {
  font-size: 2rem;
  color: #FBFBFB;
  line-height: 2.875rem;
  font-weight: 700;
}

<h2
  class="c0"
>
  heading2
</h2>
`

exports[`<Typography /> should render all headings with correct tags 3`] = `
.c0 {
  font-size: 1.5rem;
  color: #FBFBFB;
  font-weight: 500;
}

<h3
  class="c0"
>
  heading3
</h3>
`
